### PR TITLE
Adding post filter.

### PR DESCRIPTION
**Pull request by saurabhgedam**

### CODE REVIEW

File: index.php [hunk 1]
```diff
@@ -80,7 +80,22 @@
 $boolQuery->size(10);
 $boolQuery->sort(['id']);
 
-
-
-
-var_dump($boolQuery->build());
```
Comment (saurabhgedam) [inline index.php:400]: forgot to change it back to `var_dump` 🤦‍♂️

File: src/Builder/Queries/es6/PostFilterQuery.php [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    {
+        try {
+            $this->queryTreeBuilder->addArrayParam($query->build());
+        } catch (Exception $exception) {
+        }
+    }
+
+    public function build(): array
+    {
+        return $this->queryTreeBuilder->getTree();
```
Comment (BC-roliinyk) [inline src/Builder/Queries/es6/PostFilterQuery.php]: return ['post_filter [
[$this->query->build()]];

Comment (saurabhgedam) [reply src/Builder/Queries/es6/PostFilterQuery.php]: done 👍

File: src/Builder/Factory/Es6QueryFactory.php [hunk 3]
```diff
@@ -37,11 +38,14 @@ public function getQueryObject(string $queryType, string $entityType = null): Qu
...
             case 'HasParentQuery':
                 return new HasParentQuery();
+            case 'PostFilterQuery':
+                return new PostFilterQuery();
             case 'PrefixQuery':
                 return new PrefixQuery();
             default:
                 throw new \Exception('Query Class' . $queryType . 'not found');
         }
+        return new BoolQuery($queryTreeBuilder);
```
Comment (BC-roliinyk) [inline src/Builder/Factory/Es6QueryFactory.php]: probably this need to be removed?

Comment (saurabhgedam) [reply src/Builder/Factory/Es6QueryFactory.php]: yep. on it.

Comment (saurabhgedam) [reply src/Builder/Factory/Es6QueryFactory.php]: done 👍

**Pull request merged**